yt/frontends/stream/data_structures.py

```diff
@@ -1216,6 +1219,118 @@ def load_particles(data, length_unit = None, bbox=None,
 
     return sds
 
+def load_as_sph(ds, ptype, n_neighbors=64, smoothing_length=None, density=None,
+                force=False, fields=None):
+    r"""Load one particle type from a dataset as SPH particles.
+
+    This is a convenience function to create a stream particle dataset from
+    the specified particle type of an existing dataset. At a minimum,
+    "particle_position", and "particle_mass" should be available from the
+    dataset. Smoothing length and density fields would be computed if not
+    provided.
+
+    Parameters
+    ----------
+    ds : Dataset
+        The dataset to extract data from.
+    ptype : str
+        The particle type to consider.
+    n_neighbors : int
+        The number of neighbors to consider when computing smoothing length.
+    smoothing_length, density : YTArray
+        Optional SPH fields to provide. If provided, no computation would be
+        done.
+    force : bool
+        Force computation of SPH fields even when they are on-disk.
+    fields : list
+        List of on-disk fields to add.
+
+    Returns
+    -------
+    StreamParticlesDatasets
+        The constructued dataset with SPH particles.
+
+    Examples
+    --------
+
+    >>> ds_sph = load_as_sph(ds, 'PartType1')
+
+    """
+
+    l_unit = 'code_length'
+    m_unit = 'code_mass'
+    d_unit = 'code_mass / code_length**3'
+
+    ad = ds.all_data()
```

Comment:
It might be worthwhile to make it so a data object can be specified as a kwarg, and default to `all_data`.  I can see a use case for people wanting to create an sph dataset from a subset of the overall dataset.

Reply:
Thanks for the suggestion. I've changed the API.